### PR TITLE
Real Time Clock support and examples

### PR DESCRIPTION
**Pull request by Remus7**
### Pull Request Overview

This pull request adds Real Time Clock support for libtock-c apps. Specifically, in libtock folder were added rtc.c and rtc,h, which communicate with the date_time capsule from another pull requst to the main tock repository (https://github.com/tock/tock/pull/3559). Additionally,  in the examples folder was added an example rtc app, which firs sets a time and date at the beggining and then prints updated time every second.

### Testing Strategy

This pull request was tested by running the example project provided on the nucleo-stm429zi and raspberry pi pico boards.

### CODE REVIEW

File: libtock/rtc.h
Comment (ppannuto) [inline libtock/rtc.h:62]:
```suggestion
int set_date(const struct Date *set_date);
```

File: examples/rtc_app/main.c
Comment (ppannuto) [inline examples/rtc_app/main.c:18]:
```suggestion
  set_date(&date);
```

File: libtock/rtc.c
Comment (ppannuto) [inline libtock/rtc.c:43]:
```suggestion
int set_date(const struct Date *set_date){
  date = set_date->year * (1 << 9) + set_date->month * (1 << 5) + set_date->day;
  time = set_date->day_of_week * (1 << 17) + set_date->hour * (1 << 12) + set_date->minute * (1 << 6) + set_date->seconds;
```

**Review by ppannuto — CHANGES_REQUESTED** [review]
It's not a huge deal, but probably better to not have an idiom of large(?) struct-copying floating around libtock (said, without checking if we are doing that elsewhere that we shouldn't be..).

Otherwise this looks good, and is just blocked on the kernel interface settling / being merged.

**Comment by Remus7** [comment]
Thank you for the suggestions, I have merged the changes into the project. However, this week I will be unable to test whether this program still runs as expected, since I do not have access to a board with date_time support. I will try to test this again around next week, but otherwise, the requested changes have been updated.

**Comment by Remus7** [comment]
I checked the example project now, and it works as intended, including the changes that were made to the main tock capsule

**Review by ppannuto — DISMISSED** [review]
Great, this is ready to go once the kernel side is merged. We generally defer libtock merges until the kernel merge lands.

**Comment by Remus7** [comment]
The changes to the main tock repository have been merged.

**Review by ppannuto — APPROVED** [review]

**Pull request merged**
